cirq-core/cirq/transformers/target_gatesets/compilation_target_gateset.py

```diff
@@ -115,3 +115,95 @@ def postprocess_transformers(self) -> List['cirq.TRANSFORMER']:
             drop_negligible.drop_negligible_operations,
             drop_empty_moments.drop_empty_moments,
         ]
+
+
+class TwoQubitCompilationTargetGateset(CompilationTargetGateset):
+    """Abstract base class to create two-qubit target gatesets.
+
+    This base class can be used to create two-qubit compilation target gatesets. It automatically
+    implements the logic to
+
+        1. Expand composite gates acting on > 2 qubits using
+        2. Merge connected components of 1 and 2 qubit unitaries into tagged circuit operations.
+        3. Use `_decompose_single_qubit_operation` and `_decompose_two_qubit_operation` to figure
```

Comment:
If we expect users of this to interface to atleast have to override `_decompose_two_qubit_operation` we should probably make it public ?

Reply:
I think we should keep it private because `_decompose_two_qubit_operation` is not supposed to be accessed from outside the class. The docstrings do specify that `self._decompose_two_qubit_operation`  is an abstract method and should be overwritten by the users. I don't see why we need to make it public.